### PR TITLE
Addressing v0.8.0 API Review feedback

### PR DESCRIPTION
**Pull request by robscott**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
This addresses feedback from #2245.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: apis/v1beta1/gateway_types.go [hunk 1]
```diff
@@ -70,13 +70,20 @@ type GatewaySpec struct {
...
 	// 1. Port: 80, Protocol: HTTP
 	// 2. Port: 443, Protocol: HTTPS
 	//
-	// Port and protocol combinations not in this list are considered Extended.
+	// Within the TLS Conformance Profile, the below combinations of port and
+	// protocol are considered Core and MUST be supported:
+	//
+	// 1. Port: 443, Protocol: TLS
+	//
+	// Port and protocol combinations not listed above are considered Extended.
```
Comment (robscott) [inline apis/v1beta1/gateway_types.go:86]: Does this update make sense?

/cc @shaneutt @mlavacca

File: apis/v1beta1/shared_types.go [hunk 2]
```diff
@@ -25,7 +25,7 @@ import (
 // with "Core" support:
 //
 // * Gateway (Gateway conformance profile)
-// * Service (Mesh conformance profile)
+// * Service (Mesh conformance profile, experimental, ClusterIP Services only)
```
Comment (robscott) [inline apis/v1beta1/shared_types.go:28]: Does this update line up with what we're targeting for GAMMA?

/cc @howardjohn @keithmattix @kflynn @mikemorris

Comment (kflynn) [reply apis/v1beta1/shared_types.go:28]: I believe so, yes.

File: apis/v1beta1/gateway_types.go (hunk 1 above)
Comment (youngnick) [reply apis/v1beta1/gateway_types.go:86]: That makes sense to me.

**Review by youngnick — COMMENTED** [review]
This resolves everything I can remember.

/lgtm

**Review by keithmattix — APPROVED** [review]
LGTM from the GAMMA side

**Pull request merged**